File: app/src/main/java/com/savvasdalkitsis/gameframe/draw/model/FillOvalTool.java
package com.savvasdalkitsis.gameframe.draw.model;

import com.savvasdalkitsis.gameframe.grid.model.Grid;

public class FillOvalTool extends OvalTool {

    private final FillTool fillTool = new FillTool();

    @Override
    void drawOnScratch(Grid scratch, int startColumn, int startRow, int column, int row, int color) {
        super.drawOnScratch(scratch, startColumn, startRow, column, row, color);

        int rx = Math.abs(startColumn - column) / 2;
        int ry = Math.abs(startRow - row) / 2;
        int cx = Math.min(startColumn, column) + rx;
        int cy = Math.min(startRow, row) + ry;
        fillTool.fill(scratch, cx, cy, color, scratch.getColor(column, row));
    }
}

File: app/src/main/java/com/savvasdalkitsis/gameframe/feature/workspace/element/tools/model/FillOvalTool.kt
package com.savvasdalkitsis.gameframe.feature.workspace.element.tools.model

import com.savvasdalkitsis.gameframe.feature.workspace.element.grid.model.Grid

class FillOvalTool : OvalTool() {

    private val fillTool = FillTool()

    override fun drawOnScratch(scratch: Grid, startColumn: Int, startRow: Int, column: Int, row: Int, color: Int) {
        super.drawOnScratch(scratch, startColumn, startRow, column, row, color)

        val rx = Math.abs(startColumn - column) / 2
        val ry = Math.abs(startRow - row) / 2
        val cx = Math.min(startColumn, column) + rx
        val cy = Math.min(startRow, row) + ry
        fillTool.fill(scratch, cx, cy, color, scratch.getColor(column, row))
    }
}
